fastfield_codecs/src/column.rs

```diff
@@ -137,60 +137,76 @@ where V: AsRef<[T]> + ?Sized
     }
 }
 
-struct MonotonicMappingColumn<C, T, Input> {
+struct MonotonicMappingColumn<C, T, U, Input> {
     from_column: C,
-    monotonic_mapping: T,
+    monotonic_mapping_to_output: T,
+    monotonic_mapping_to_input: U,
```

Comment:
because of the naming I thought you wanted to build something like 

```
g(col(f(i)))
```
Can you call it maybe
`monotonic_mapping` and
`monotonic_mapping_inv`

Also... note there is something tricky here. By introducing the reciprocal function, you forbid the use of non-strictly monotonic function.

Reply:
We need that for
`fn get_between_vals(&self, range: RangeInclusive<T>) -> Vec<u64> `

The user provides the original values and we need to monotonic map them in the same way the serialization does before calling the underlying column.

> By introducing the reciprocal function, you forbid the use of non-strictly monotonic function.

We had this limitation before, but only implicit, via serialization and reading. I'll move the code together to make that more explicit.

The API is a little strange for serialization, since semantically it doesn't make sense to have an inverse. In that case it's a one way and the inverse is unimplemented.


yes, `monotonic_mapping` and `monotonic_mapping_inv` are better names. Input and output flip depending if we are serializing or reading, and that's gets confusing with the mapping.